extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -66,6 +66,26 @@ public static function onBeforeInitialize( Title $title, $unused, OutputPage $ou
 		return true;
 	}
 
+	/**
+	 * Remove Content-Security-Policy header when downgrading HTTPS to HTTP
+	 * Microsoft Edge starting from v17 misinterprets Content-Security-Policy: upgrade-insecure-requests
+	 * during redirects and falls into a loop HTTPS -> HTTP -> HTTPS -> ...
+	 *
+	 * @param $out
+	 * @param $redirect
+	 * @param $code
+	 * @param $redirectedBy
+	 */
+	public static function onBeforePageRedirect( $out, &$redirect, &$code, &$redirectedBy ) {
+		$currentUri = \WikiFactoryLoader::getCurrentRequestUri( $_SERVER, false, true );
+		$currentProtocol = parse_url( $currentUri, PHP_URL_SCHEME );
```

Comment:
I think you can use `WebRequest::detectProtocol()` if you don't need the full uri

Reply:
Yup, that works too, thanks!